ims_python_helper/__init__.py

```diff
@@ -284,16 +363,25 @@ def get_empty_image_record_for_name(self, image_name: str, skip_existing: bool)
                 has been uploaded, and `skip_existing` is True
         """
         if skip_existing:
+            matching_uploaded_images = []
+            matching_empty_images = []
             try:
                 existing_images = self._ims_images_get()
                 for image in existing_images:
                     if image.get("name") == image_name:
                         if image.get("link"):
-                            raise ImsImageAlreadyUploaded(image)
+                            matching_uploaded_images.append(image)
                         else:
                             LOGGER.info("Image \"%s\" with ID \"%s\" found, but it has not been "
                                         "uploaded yet.", image_name, image.get("id"))
-                            return image
+                            matching_empty_images.append(image)
+                if matching_uploaded_images:
+                    raise ImsImagesExistWithName(image_name, matching_uploaded_images)
+                if matching_empty_images:
+                    empty_image = matching_empty_images.pop()
+                    LOGGER.info("Uploading image artifacts to empty image %s",
```

Comment:
Quote image ID for consistency with other existing logged messages.

```suggestion
                    LOGGER.info("Uploading image artifacts to empty image with ID \"%s\"",
```